cosmo_tester/test_suites/test_security/secured_nodecellar_test.py

```diff
@@ -0,0 +1,113 @@
+    import OpenStackNodeCellarTestBase
+
+from cloudify_rest_client.client import CloudifyClient
+from cloudify_cli.constants import (CLOUDIFY_USERNAME_ENV,
+                                    CLOUDIFY_PASSWORD_ENV)
+
+TEST_CFY_USERNAME = 'user1'
+TEST_CFY_PASSWORD = 'pass1'
+
+
+class SecuredOpenstackNodecellarTest(OpenStackNodeCellarTestBase):
+
+    def test_secured_openstack_nodecellar(self):
+        inputs_path, mb_path = self._copy_manager_blueprint()
+        self.test_manager_blueprint_path = path(mb_path)
+        self.test_inputs_path = path(inputs_path)
+
+        self._update_manager_blueprint_security()
+        self._bootstrap()
+        self._set_credentials_env_vars()
+        self._running_env_setup()
+
+        self._test_openstack_nodecellar('openstack-blueprint.yaml')
+
+    def _copy_manager_blueprint(self):
+        return util.generate_unique_configurations(
+            workdir=self.workdir,
+            original_inputs_path=self.env.cloudify_config_path,
+            original_manager_blueprint_path=self.env._manager_blueprint_path)
+
+    def _update_manager_blueprint_security(self):
+        security_path = 'node_templates.manager.properties.cloudify.security'
+        security_settings = {
+            'enabled': 'true',
+            'userstore_driver': {
+                'implementation':
+                    'flask_securest.userstores.file:FileUserstore',
+                'properties': {
+                    'userstore_file': 'users.yaml',
+                    'identifying_attribute': 'username'
+                }
+            },
+            'authentication_methods': [
+                {
+                    'name': 'password',
+                    'implementation': 'flask_securest.'
+                                      'authentication_providers.password:'
+                                      'PasswordAuthenticator',
+                    'properties': {
+                        'password_hash': 'plaintext'
+                    }
+                },
+                {
+                    'name': 'token',
+                    'implementation': 'flask_securest.'
+                                      'authentication_providers.token:'
+                                      'TokenAuthenticator',
+                    'properties': {
+                        'secret_key': 'yaml_secret'
+                    }
+                }
+            ]
+        }
+        self._update_manager_blueprint(security_path, security_settings)
+
+    def _update_manager_blueprint(self, prop_path, new_value):
+        with util.YamlPatcher(self.test_manager_blueprint_path) as patch:
+            patch.set_value(prop_path, new_value)
+
+    def _bootstrap(self):
+        self.cfy.bootstrap(blueprint_path=self.test_manager_blueprint_path,
+                           inputs_file=self.test_inputs_path,
+                           task_retries=5,
+                           install_plugins=False)
```

Comment:
Let's discuss `install_plugins` tomorrow, I also have a bug in the rest_plugins test where it is also `False`.
Generally, when running tests locally, you do want this to be `False` so that it doesn't override packages installed in your environment, but when run as part of the automation, some plugins may not be installed (such as the fabric plugin) so the install_plugins is required there.